syntax = "proto3";
import "common.proto";
package autodrive.workflow.proto;

enum CameraID {
  CAMERA_DIRECTION_UNKNOWN = 0;
  CAMERA_DIRECTION_BACK = 1;
  CAMERA_DIRECTION_FRONT = 2;
  CAMERA_DIRECTION_LEFT = 3;
  CAMERA_DIRECTION_RIGHT = 4;
  CAMERA_DIRECTION_PIN_REAR = 5;
  CAMERA_DIRECTION_PIN_FRONT = 6;
  CAMERA_DIRECTION_OMNI_BACK = 7;
  CAMERA_DIRECTION_OMNI_FRONT = 8;
  CAMERA_DIRECTION_OMNI_LEFT = 9;
  CAMERA_DIRECTION_OMNI_RIGHT = 10;
}

message Extrinsic {
  float x = 1;
  float y = 2;
  float z = 3;
  float roll = 4;
  float pitch = 5;
  float yaw = 6;
  CameraID cam_id = 7;
  uint32 num_extrinsic = 8;
}

message CameraExtrinsicsMsg {
  // Header
	MsgHeader header = 1;
  // Error
	Error error = 2;
  // Extrinsics
  repeated Extrinsic extrinsics = 3;
}
